pkg/ansible/runner/runner.go

```diff
@@ -310,7 +310,7 @@ func (r *runner) makeParameters(u *unstructured.Unstructured) map[string]interfa
 	parameters := paramconv.MapToSnake(spec)
 	parameters["meta"] = map[string]string{"namespace": u.GetNamespace(), "name": u.GetName()}
 
-	objKey := fmt.Sprintf("_%v_%v", strings.Replace(r.GVK.Group, ".", "_", -1),
+	objKey := fmt.Sprintf("_%v_%v", strings.ReplaceAll(strings.ReplaceAll(r.GVK.Group, ".", "_"), "-", "_"),
```

Comment:
could you put a comment above this line? maybe even including an example. And what david said about the test would be nice.

Reply:
Are `.` and `-` the only characters than need replacing? Regardless this might be easier to read as a regexp and `re.ReplaceAllString`.

Reply:
broke it out into a function which should be more clear. It should be easy to add more characters if needed, as well as reusing it on additional parameters in the future if that ever becomes a concern (might be worth looking into whether we need to do that, but wanted to keep scope here relatively limited)